ui/app/src/services/content.ts

```diff
@@ -1482,25 +1483,31 @@ export function cut(site: string, item: SandboxItem): Observable<any> {
 
 export function paste(
   site: string,
-  item: SandboxItem
+  path: string
 ): Observable<{ site: string; status: string[] }> {
   return get(
-    `/studio/api/1/services/api/1/clipboard/paste-item.json?site=${site}&parentPath=${item.path}`
+    `/studio/api/1/services/api/1/clipboard/paste-item.json?site=${site}&parentPath=${path}`
   ).pipe(pluck('response'), catchError(errorSelectorApi1));
 }
 
 export function duplicate(
   site: string,
-  item: SandboxItem,
-  parentItem: SandboxItem
-): Observable<SandboxItem> {
+  path: string
+): Observable<string> {
+  let parentPath: any = path;
+  if (path.endsWith('index.xml')) {
+    parentPath = withoutIndex(path);
+  }
+  parentPath = parentPath.split('/');
+  parentPath.pop();
+  parentPath = parentPath.join('/');
   return forkJoin({
-    copy: copy(site, item),
-    newItem: paste(site, parentItem)
-  }).pipe(map(({ copy, newItem }) => ({ ...item, path: newItem.status[0] })));
+    copy: copy(site, path),
+    newItem: paste(site, parentPath)
+  }).pipe(map(({ copy, newItem }) => newItem.status[0]));
 }
 
-export function getPages(site: string, item: any): Observable<any> {
+export function getPages(site: string, item: DetailedItem): Observable<any> {
```

Comment:
- Return type
- Receive the path instead of the full item
- What's this for?

Reply:
this is used for CopyDialog, its returns the copy item and his childrens